src/webpub_manifest_parser/core/parsers.py

```diff
@@ -0,0 +1,751 @@
+        :rtype: Any
+
+        :raise: ValidationError
+        """
+        return value
+
+    def parse(self, value):
+        """Parse the value according to the specified format.
+
+        :param value: Value
+        :type value: Any
+
+        :return: Parsed value
+        :rtype: Any
+
+        :raise: ValidationError
+        """
+        self._validate(value)
+        result = self._parse(value)
+
+        return result
+
+
+class URIParser(FormatChecker):
+    """URI parser."""
+
+    def __init__(self):
+        """Initialize a new instance of URIParser class."""
+        super(URIParser, self).__init__("uri")
+
+    def _parse(self, value):
+        """Parse the URI into a Python dictionary containing URI's subcomponents.
+
+        :param value: Value
+        :type value: Any
+
+        :return: URI
+        :rtype: str
+        """
+        return value
+
+
+class URITemplateParser(StringParser):
+    """URI template parser."""
+
+    def parse(self, value):
+        """Parse the URI into a Python dictionary containing URI's subcomponents.
+
+        :param value: Value
+        :type value: Any
+
+        :return: URI template
+        :rtype: URITemplate
+        """
+        URITemplate(value)
+
+        return value
+
+
+class URIReferenceParser(FormatChecker):
+    """URI-reference parser."""
+
+    def __init__(self):
+        """Initialize a new instance of URIReferenceParser class."""
+        super(URIReferenceParser, self).__init__("uri-reference")
+
+    def _parse(self, value):
+        """Parse the URI-reference into a Python dictionary containing URI's subcomponents.
+
+        :param value: Value
+        :type value: Any
+
+        :return: URI-reference
+        :rtype: str
+        """
+        return value
+
+
+class DateParser(FormatChecker):
+    """Date parser."""
+
+    def __init__(self):
+        """Initialize a new instance of DateParser class."""
+        super(DateParser, self).__init__("date")
+
+    def _parse(self, value):
+        """Parse a date & time string into datetime object.
+
+        :param value: Value
+        :type value: Any
+
+        :return: Parsed date object
+        :rtype: datetime.datetime
+        """
+        return datetime.datetime.strptime(value, "%Y-%m-%d")
+
+
+class DateTimeParser(FormatChecker):
+    """Date & time parser."""
+
+    def __init__(self):
+        """Initialize a new instance of DateTimeParser class."""
+        super(DateTimeParser, self).__init__("date-time")
+
+    def _parse(self, value):
+        """Parse a date & time string into datetime object.
+
+        :param value: Value
+        :type value: Any
+
+        :return: Parsed date & time object
+        :rtype: datetime.datetime
+        """
+        timestamp = strict_rfc3339.rfc3339_to_timestamp(value)
+
+        return datetime.datetime.utcfromtimestamp(timestamp)
+
+
+class ArrayParser(ValueParser):
+    """Array parser."""
+
+    def __init__(self, item_parser, unique_items=False):
```

Comment:
Mickaël mentioned the idea of treating some arrays as sets. "Array with a uniqueness constraint" is slightly different from "set", but they're pretty similar and I don't think the distinction matters in an OPDS 2 context. So this might be a good way to implement sets -- or this might be code that should be moved over to a new `SetParser` class.

Reply:
OK, I see from your comments that you had a similar idea.

Reply:
I'm just thinking whether we really need `SetParser` because my first intention was to name parsers after the types and formats in the spec. So there is `array` type and there is `ArrayParser`

Reply:
Now that you mention this, uniqueness is not specified in the JSON schema. But I don't think it makes sense to have duplicate relations anyway.

Reply:
@mickael-menu-mantano, actually JSON schema contains `uniqueItems`:
```
"links": {
      "description": "Feed-level links such as search or pagination",
      "type": "array",
      "items": {
        "$ref": "https://readium.org/webpub-manifest/schema/link.schema.json"
      },
      "uniqueItems": true
```

Reply:
For `links` yeah but not for `rel`.